src/main/scala/com/databricks/labs/mosaic/expressions/raster/RST_FromContent.scala

```diff
@@ -0,0 +1,136 @@
+import com.databricks.labs.mosaic.core.types.RasterTileType
+import com.databricks.labs.mosaic.core.types.model.MosaicRasterTile
+import com.databricks.labs.mosaic.datasource.gdal.ReTileOnRead
+import com.databricks.labs.mosaic.expressions.base.{GenericExpressionFactory, WithExpressionInfo}
+import com.databricks.labs.mosaic.functions.MosaicExpressionConfig
+import com.databricks.labs.mosaic.utils.PathUtils
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.analysis.FunctionRegistry.FunctionBuilder
+import org.apache.spark.sql.catalyst.expressions.codegen.CodegenFallback
+import org.apache.spark.sql.catalyst.expressions.{CollectionGenerator, Expression, Literal, NullIntolerant}
+import org.apache.spark.sql.types.{DataType, IntegerType, StringType, StructField, StructType}
+import org.apache.spark.unsafe.types.UTF8String
+
+import java.nio.file.{Files, Paths, StandardOpenOption}
+
+/**
+  * The raster for construction of a raster tile. This should be the first
+  * expression in the expression tree for a raster tile.
+  */
+case class RST_FromContent(
+    rasterExpr: Expression,
+    driverExpr: Expression,
+    parentPathExpr: Expression,
+    sizeInMB: Expression,
+    expressionConfig: MosaicExpressionConfig
+) extends CollectionGenerator
+      with Serializable
+      with NullIntolerant
+      with CodegenFallback {
+
+    override def dataType: DataType = RasterTileType(expressionConfig.getCellIdType)
+
+    protected val geometryAPI: GeometryAPI = GeometryAPI.apply(expressionConfig.getGeometryAPI)
+
+    protected val indexSystem: IndexSystem = IndexSystemFactory.getIndexSystem(expressionConfig.getIndexSystem)
+
+    protected val cellIdDataType: DataType = indexSystem.getCellIdDataType
+
+    override def position: Boolean = false
+
+    override def inline: Boolean = false
+
+    override def children: Seq[Expression] = Seq(rasterExpr, driverExpr, parentPathExpr, sizeInMB)
+
+    override def elementSchema: StructType = StructType(Array(StructField("tile", dataType)))
+
+    /**
+      * subdivides raster binary content into tiles of the specified size (in MB).
+      * @param input
+      *   The input file path.
+      * @return
+      *   The tiles.
+      */
+    override def eval(input: InternalRow): TraversableOnce[InternalRow] = {
+        GDAL.enable(expressionConfig)
+        //parentPath may be null (it is not used here as content may be different)
+        val parentPath = {
+            try {
+                parentPathExpr.eval(input).asInstanceOf[UTF8String].toString
+            } catch {
+                case _: Any => null
+            }
+        }
+        val driver = driverExpr.eval(input).asInstanceOf[UTF8String].toString
+        val ext = GDAL.getExtension(driver)
+        var rasterArr = rasterExpr.eval(input).asInstanceOf[Array[Byte]]
+        val targetSize = sizeInMB.eval(input).asInstanceOf[Int]
+        if (targetSize <= 0 && rasterArr.length <= Integer.MAX_VALUE) {
```

Comment:
@mjohns-databricks there is no need to check for `Integer.MAX_VALUE` no binary column in spark can be larger than this anyhow. This is where 2GB limit comes from.